blockchain-btc/src/main/java/io/mywish/btc/blockchain/BtcBCModule.java

```diff
@@ -86,16 +89,45 @@ public BtcNetwork btcNetTest(
 
     @ConditionalOnBean(name = NetworkType.BTC_MAINNET_VALUE)
     @Bean
-    public BtcScanner btcScannerMain(
-            final @Qualifier(NetworkType.BTC_MAINNET_VALUE) BtcNetwork network,
+    public LastBlockPersister btcMainnetLastBlockPersister(
+            final @Value("${etherscanner.bitcoin.db-block-persister:#{false}}") boolean isDbPersister,
+            final BtcNetwork network,
+            final LastBlockRepository lastBlockRepository,
+            final @Value("${etherscanner.start-block-dir}") String dir,
+            final @Value("${etherscanner.bitcoin.last-block.mainnet:#{null}}") Long lastBlock
+
+    ) {
+        return isDbPersister
+                ? new LastBlockDbPersister(network.getType(), lastBlockRepository, lastBlock)
+                : new LastBlockFilePersister(network.getType(), dir, lastBlock);
```

Comment:
it looks like bad idea to use imperative condition there. I think it's better to define both db and file persisters like separately beans, but disable on of them ConditionOnProperty